src/scriptdev2/scripts/kalimdor/durotar.cpp

```diff
@@ -33,89 +33,185 @@ EndContentData */
         {
-            m_creature->RemoveAurasDueToSpell(SPELL_PEON_SLEEP);
+            DoScriptText(SAY_PEON_AWOKEN, m_creature, pInvoker);
+            m_creature->PlayDistanceSound(urand(0, 1) ? SOUND_PEON_AWAKE_1 : SOUND_PEON_AWAKE_2);
+            m_creature->SetWalk(false); //peon will run if awoken by player
+            ((Player*)pInvoker)->KilledMonsterCredit(m_creature->GetEntry(), m_creature->GetObjectGuid());
+        }
+        else
+            m_creature->SetWalk(true); //peon will walk if awoken on its own
 
+        m_creature->RemoveAurasDueToSpell(SPELL_PEON_SLEEP);
+        RestartSleepTimer();
+        GoLumberPile();
+    }
+
+    void GoLumberPile()
+    {
+        if (GameObject* pLumber = GetClosestGameObjectWithEntry(m_creature, GO_LUMBERPILE, 15.0f))
+        {
             float fX, fY, fZ;
-            m_creature->SetWalk(false);
             pLumber->GetContactPoint(m_creature, fX, fY, fZ, CONTACT_DISTANCE);
-
-            if (pInvoker->GetTypeId() == TYPEID_PLAYER)
-            {
-                DoScriptText(SAY_PEON_AWAKE_1, m_creature);
-                ((Player*)pInvoker)->KilledMonsterCredit(m_creature->GetEntry(), m_creature->GetObjectGuid());
-                m_creature->GetMotionMaster()->MovePoint(1, fX, fY, fZ);
-            }
-            else
-                m_creature->GetMotionMaster()->MovePoint(2, fX, fY, fZ);
+            m_creature->HandleEmote(EMOTE_STATE_NONE); //cancel chopping
+            m_creature->GetMotionMaster()->MovePoint(1, fX, fY, fZ);
+            m_uiKneelTimer = 2500; //timer duration guessed, true retail time unknown
         }
         else
             script_error_log("No GameObject of entry %u was found in range or something really bad happened.", GO_LUMBERPILE);
     }
 
     void MovementInform(uint32 uiMotionType, uint32 uiPointId) override
     {
-        if (uiMotionType != POINT_MOTION_TYPE || !uiPointId)
-            return;
-
-        m_creature->HandleEmote(EMOTE_STATE_WORK_CHOPWOOD);
-        // TODO - random bevahior for self-invoked awakening guesswork
-        m_uiResetSleepTimer = uiPointId == 1 ? 80000 : urand(30000, 60000);
+        if (uiMotionType == POINT_MOTION_TYPE && uiPointId)
+        {
+            switch (uiPointId)
+            {
+                case 1: //lumber pile
+                {
+                    m_creature->SetWalk(true);
+                    m_bIsAtPile = true;
+                    break;
+                }
+                case 2: //spawn
+                {
+                    m_creature->GetMotionMaster()->MoveIdle();
+                    m_creature->HandleEmote(EMOTE_STATE_NONE);
+                    m_creature->SetFacingTo(m_fSpawnO);
+                    m_creature->CastSpell((Unit*)m_creature, SPELL_PEON_SLEEP, false);
+                    RestartWakeTimer();
+                    break;
+                }
+            }
+        }
+        else if (uiMotionType == WAYPOINT_MOTION_TYPE)
```

Comment:
I didn't say it's wrong. I was asking you if that's your intended purpose.
Usually for WAYPOINT_MOTION_TYPE we provide a uiPointId.